qa/common/gen_qa_implicit_models.py

```diff
@@ -690,17 +689,236 @@ def create_plan_fixed_rf_modelfile(models_dir, model_version, max_batch, dtype,
+    for i in shape:
+        if i == -1:
+            min_shape = min_shape + [1]
+            opt_shape = opt_shape + [8]
+            max_shape = max_shape + [32]
+        else:
+            min_shape = min_shape + [i]
+            opt_shape = opt_shape + [i]
+            max_shape = max_shape + [i]
+
+    profile = builder.create_optimization_profile()
+    profile.set_shape("INPUT", min_shape, opt_shape, max_shape)
+    profile.set_shape("INPUT_STATE", min_shape, opt_shape, max_shape)
+    if max_batch != 0:
+        profile.set_shape("START", [1] + unit_shape, [max_batch] + unit_shape,
+                          [max_batch] + unit_shape)
+        profile.set_shape("READY", [1] + unit_shape, [max_batch] + unit_shape,
+                          [max_batch] + unit_shape)
+    else:
+        profile.set_shape("START", unit_shape, unit_shape, unit_shape)
+        profile.set_shape("READY", unit_shape, unit_shape, unit_shape)
+
+    config = builder.create_builder_config()
+    config.flags = flags
+    config.add_optimization_profile(profile)
+
+    config.max_workspace_size = 1 << 20
+    try:
+        engine_bytes = builder.build_serialized_network(network, config)
+    except AttributeError:
+        engine = builder.build_engine(network, config)
+        engine_bytes = engine.serialize()
+        del engine
+
+    model_name = tu.get_sequence_model_name(
+        "plan_nobatch" if max_batch == 0 else "plan", dtype)
+    model_version_dir = models_dir + "/" + model_name + "/" + str(model_version)
+
+    try:
+        os.makedirs(model_version_dir)
+    except OSError as ex:
+        pass  # ignore existing dir
+
+    with open(model_version_dir + "/model.plan", "wb") as f:
+        f.write(engine_bytes)
+
+
 def create_plan_modelfile(models_dir, model_version, max_batch, dtype, shape):
 
     if not tu.validate_for_trt_model(dtype, dtype, dtype, shape, shape, shape):
         return
 
     if dtype != np.float32:
-        create_plan_fixed_rf_modelfile(models_dir, model_version, max_batch,
-                                       dtype, shape)
+        if (not tu.shape_is_fixed(shape)):
+            create_plan_dynamic_rf_modelfile(models_dir, model_version,
+                                             max_batch, dtype, shape)
+        else:
+            create_plan_fixed_rf_modelfile(models_dir, model_version, max_batch,
+                                           dtype, shape)
     else:
-        create_plan_fixed_modelfile(models_dir, model_version, max_batch, dtype,
-                                    shape)
+        if (not tu.shape_is_fixed(shape)):
+            create_plan_dynamic_modelfile(models_dir, model_version, max_batch,
+                                          dtype, shape)
+        else:
+            create_plan_fixed_modelfile(models_dir, model_version, max_batch,
+                                        dtype, shape)
 
 
 def create_plan_modelconfig(models_dir, model_version, max_batch, dtype, shape):
```

Comment:
Do you need to generate the config for the new dynamic shape models?

Reply:
I generated them in the DLcluster.

Reply:
I think Guan meant do you need to write a different function for creating plan_modelconfig file for variable dimension case. 
I think the same function should work as the information regarding dynamic shape is communicated via shape parameter.